ambry-frontend/src/main/java/com/github/ambry/frontend/UserQuotaRequestCostPolicy.java

```diff
@@ -66,28 +65,11 @@ private String restMethodToCostMetric(RestMethod restMethod) {
 
   /**
    * Calculate the storage cost incurred to serve a request.
-   * For post requests this is the number bytes in GB of the blob data. For all other write requests the default is DEFAULT_COST_FOR_HEAD_DELETE_TTL.
-   * For read requests there is no storage cost incurred.
    * @param restRequest {@link RestRequest} to find type of request.
-   * @param restResponseChannel {@link RestResponseChannel} object.
    * @return storage cost.
    */
-  private double calculateStorageCost(RestRequest restRequest, RestResponseChannel restResponseChannel) {
-    switch (restRequest.getRestMethod()) {
-      case POST:
-        long contentSize = 0;
-        if (restRequest.getUri().contains(Operations.STITCH)) {
-          contentSize = Long.parseLong((String) restResponseChannel.getHeader(RestUtils.Headers.BLOB_SIZE));
-        } else {
-          contentSize = restRequest.getBytesReceived();
-        }
-        return contentSize / BYTES_IN_GB;
-      case DELETE:
-      case PUT:
-        return CU_COST_UNIT / BYTES_IN_GB; // Assuming 1 chunk worth of storage cost for deletes and ttl updates.
-      default:
-        return 0;
-    }
+  private double calculateStorageCost(RestRequest restRequest) {
+    return RestUtils.isUploadRequest(restRequest) ? restRequest.getBlobBytesReceived() : 0;
```

Comment:
My thinking is that storage cost for now, should work only for uploads. This is because when we calculate the storage used for certain container, we only accumulate the PUT, not the TTL_UPDATE, not the DELETE either. (of course, PUT is the chunk content with blob metadata).
And for stitch upload, since the chunk content is the list of blob ids, it's very hard to know the size of the metadata chunk in this write path. I figured we can probably ignore it in the frontend, and let the server figure it out.